### ISSUE TITLE
[BUG] `http` plugin resests `http.Request` attributes

### ISSUE THREAD

**Issue by zaishui15** [spiral/roadrunner#789]
```go
func (s *Service) middleware(next http.HandlerFunc) http.HandlerFunc {
 return func(w http.ResponseWriter, r *http.Request) {
     r = attributes.Init(r)
     attributes.Set(r, "key", "value")
            next(w, r)
 }
}
```
```go
func (p *Plugin) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	defer func() {
		err := r.Body.Close()
		if err != nil {
			p.log.Error("body close", "error", err)
		}
	}()
	if headerContainsUpgrade(r) {
		http.Error(w, "server does not support upgrade header", http.StatusInternalServerError)
		return
	}

	if p.https != nil && r.TLS == nil && p.cfg.SSLConfig.Redirect {
		p.redirect(w, r)
		return
	}

	if p.https != nil && r.TLS != nil {
		w.Header().Add("Strict-Transport-Security", "max-age=31536000; includeSubDomains; preload")
	}

	r = attributes.Init(r)
	// protect the case, when user sendEvent Reset and we are replacing handler with pool
	p.RLock()
	p.handler.ServeHTTP(w, r)
	p.RUnlock()
}
```
When I want to set attributes, HTTP plugin resets attributes

**Comment by rustatian** [comment]
@zaishui15 Hey, will be addressed in the `v2.5.0`, thanks for the report.

**Comment by rustatian** [comment]
BTW, `middleware` func in your sample should be exported to be found and connected by the `http` plugin as a middleware.

**Comment by zaishui15** [comment]
thinks, the problem has been solved, May be fixed by #783
